pkg/core/native/native_test/cryptolib_verification_test.go

```diff
@@ -0,0 +1,879 @@
+		opcode.LDLOC4) // load pubCnt
+	emit.Int(vrf.BinWriter, int64(n))      // push n.
+	emit.Opcodes(vrf.BinWriter, opcode.GE, // pubCnt >= n
+		opcode.OR) // sigCnt >= m || pubCnt >= n
+	emit.Instruction(vrf.BinWriter, opcode.JMPIF, []byte{0}) // jump to the end of the script if (sigCnt >= m || pubCnt >= n).
+	loopConditionOffset := vrf.Len()
+
+	// Loop start. Prepare arguments and call CryptoLib's verifyWithECDsa.
+	emit.Int(vrf.BinWriter, int64(native.Secp256k1Keccak256)) // push Koblitz curve identifier.
+	emit.Opcodes(vrf.BinWriter, opcode.LDLOC0,                // load signatures.
+		opcode.LDLOC3,             // load sigCnt.
+		opcode.PICKITEM,           // pick signature at index sigCnt.
+		opcode.LDLOC1,             // load pubs.
+		opcode.LDLOC4,             // load pubCnt.
+		opcode.PICKITEM,           // pick pub at index pubCnt.
+		opcode.LDLOC2,             // load msg.
+		opcode.PUSH4, opcode.PACK) // pack 4 arguments for 'verifyWithECDsa' call.
+	emit.AppCallNoArgs(vrf.BinWriter, cryptoLibH, "verifyWithECDsa", callflag.All) // emit the call to 'verifyWithECDsa' itself.
+
+	// Update loop variables.
+	emit.Opcodes(vrf.BinWriter, opcode.LDLOC3, opcode.ADD, opcode.STLOC3, // increment sigCnt if signature is valid.
+		opcode.LDLOC4, opcode.INC, opcode.STLOC4) // increment pubCnt.
+
+	// End of the loop.
+	emit.Instruction(vrf.BinWriter, opcode.JMP, []byte{0}) // jump to the start of cycle.
+	loopEndOffset := vrf.Len()
+
+	// Return condition: the number of valid signatures should be equal to m.
+	progRetOffset := vrf.Len()
+	emit.Opcodes(vrf.BinWriter, opcode.LDLOC3)   // load sigCnt.
+	emit.Int(vrf.BinWriter, int64(m))            // push m.
+	emit.Opcodes(vrf.BinWriter, opcode.NUMEQUAL) // push m == sigCnt.
+
+	require.NoError(t, vrf.Err)
+	script := vrf.Bytes()
+
+	// Set JMP* instructions offsets. "-1" is for short JMP parameter offset. JMP parameters
+	// are relative offsets.
+	script[sigsLenCheckEndOffset-1] = byte(checkStartOffset - sigsLenCheckEndOffset + 2)
+	script[loopEndOffset-1] = byte(loopStartOffset - loopEndOffset + 2)
+	script[loopConditionOffset-1] = byte(progRetOffset - loopConditionOffset + 2)
+
+	return script
+	// Here's an example of the resulting single witness invocation script (261 bytes length, the length may vary depending on m/n):
+	// NEO-GO-VM > loadbase64 VwUAQxMoBUkJQBPAcAwhAnDdr99Ja4K3I81KURO2xs8b+dYYVIaMhbDFTYO4FCnKDCECuBwcms5bdqbWeBZ1cnMAJ8z/uUMcxnIK0CxTyxNdYqAMIQLQHl4aPx8PZOgu4EQUh0qCPaCfaZZPLNNS9ZVPcmuXpwwhA+YKTuJo6wB/u/CQdzJczfQQaMk6LHfMlSZMdBD2qCV1FMBxQcX7oOADAAAAAAEAAACeFI1BLVEIMBDOi3IQcxB0axO4bBS4kiRCABhoa85pbM5qFMAfDA92ZXJpZnlXaXRoRUNEc2EMFBv1dasRiWiEE2EKNaEohs3gtmxyQWJ9W1JrnnNsnHQiuWsTsw==
+	// READY: loaded 262 instructions
+	// NEO-GO-VM 0 > ops
+	// INDEX    OPCODE       PARAMETER
+	// 0        INITSLOT     5 local, 0 arg    <<
+	// 3        DEPTH
+	// 4        PUSH3
+	// 5        JMPEQ        10 (5/05)
+	// 7        CLEAR
+	// 8        PUSHF
```

Comment:
Can ABORT here, btw. A bit less code with ~the same effect. But can be discussed separately.

Reply:
We can, and it will reduce the invocation cost, but I don't quite like it. It will be not very pleasant for the user to see that witness check script FAULTed:
`transaction 3c17b0a88c6fa6af674d8a58a6648b1878bf07012592fcfb7c34e81fa36e21c2 failed to verify: witness #0: signature check failed: vm execution has failed: at instruction 7 (ABORT): ABORT`
However, `System.Crypto.CheckMultisig` does exactly the same thing, so let's take ABORT.

Fixed.